kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -65,11 +68,14 @@
     @Getter
     protected final KafkaServiceConfiguration kafkaConfig;
 
+    private final ScheduledExecutorService executor;
```

Comment:
The executor will be created for each connection. Should we use a common executor service across all connections?

Reply:
If we want to use a common thread pool, different channels may affect each other.
If we are sure to use a common thread pool, we need to use OrderedScheduler to ensure the order of the same channel;

Reply:
One thread per connection is not a good design. If we did so, KoP might not be able to support a lot of connections.

The `writeAndFlushResponseToClient` method just retrieves completed requests from `requestQueue` and send them. I think the influence for other channels is not significant.

And we can also use a round robin algorithm to assign a thread to channels. For example, if the executor is a fixed thread pool with 20 threads, the assignment could be
- Thread 0 for channel 0, 20, 40, ...
- Thread 1 for channel 1, 21, 41, ...
- ...

Reply:
You are correct, I will use a public order excutor to process the send response。

Reply:
I have another question. Since the thread only calls `writeAndFlushResponseToClient`, is it necessary to use an ordered executor? Because the `requestQueue` field belongs to each channel.

Reply:
IMO, in order to ensure order, we need to use order excuter